### PR TITLE
feat: implementa módulo de Reavaliações do paciente

### LINKED ISSUE #67: Implementar Reavaliações do paciente no frontend
## Descrição do problema ou necessidade

Com a implementação do módulo de **Reavaliações do paciente** no backend, conforme a issue [FabioCarlesso/carlessopilatesapi#54](https://github.com/FabioCarlesso/carlessopilatesapi/issues/54), o frontend precisa permitir que o usuário visualize, cadastre, edite e consulte reavaliações vinculadas a um paciente.

O prontuário deve oferecer uma interface clara para registrar reavaliações periódicas, permitindo acompanhar a evolução clínica do paciente, comparar resultados anteriores e apoiar ajustes no plano de tratamento.

## Objetivo da atividade

Implementar no frontend o fluxo de **Reavaliações do paciente**, integrando com os endpoints disponibilizados pelo backend, permitindo o gerenciamento das reavaliações dentro do contexto do prontuário/paciente.

## Escopo da implementação

- Criar estrutura frontend para o módulo de reavaliações do paciente.
- Criar ou ajustar rotas relacionadas ao paciente/prontuário para acessar as reavaliações.
- Implementar tela/listagem de reavaliações vinculadas a um paciente.
- Implementar formulário para criação de nova reavaliação.
- Implementar formulário ou fluxo para edição de reavaliação existente.
- Implementar visualização/detalhamento de uma reavaliação.
- Criar service Angular para comunicação com a API de reavaliações.
- Criar models/interfaces TypeScript para request e response de reavaliação.
- Tratar estados de carregamento, sucesso e erro nas operações.
- Validar campos obrigatórios no formulário.
- Permitir múltiplas reavaliações por paciente.
- Garantir que a reavaliação esteja sempre associada ao paciente correto.

### Campos mínimos sugeridos para o formulário

- Data da reavaliação.
- Comparativo com avaliação anterior.
- Evolução da dor.
- Evolução de força.
- Evolução de mobilidade.
- Evolução funcional.
- Objetivos alcançados.
- Pontos de atenção.
- Ajustes recomendados no plano de tratamento.
- Observações gerais.

### Integração esperada com o backend

Considerar os endpoints definidos na implementação da issue do backend:

- Criar reavaliação para um paciente.
- Buscar reavaliação por ID.
- Listar reavaliações de um paciente.
- Atualizar reavaliação.

Ajustar nomes de rotas, payloads e responses conforme a implementação final do backend.

## Critérios de aceite

- Deve ser possível acessar as reavaliações a partir do contexto de um paciente.
- Deve ser possível listar todas as reavaliações cadastradas para um paciente.
- Deve ser possível criar uma nova reavaliação para um paciente existente.
- Deve ser possível consultar os detalhes de uma reavaliação específica.
- Deve ser possível editar uma reavaliação existente.
- O formulário deve validar os campos obrigatórios antes de enviar para a API.
- O sistema deve exibir mensagens adequadas de sucesso e erro.
- O sistema deve tratar erro ao tentar carregar reavaliações de paciente inexistente ou inválido.
- A data da reavaliação deve ser exibida e enviada corretamente.
- A implementação deve seguir o padrão visual, estrutural e de organização já utilizado no frontend.
- A integração deve utilizar services Angular, evitando chamadas HTTP diretamente nos componentes quando não for padrão do projeto.

## Sugestão de validação/testes

- Validar criação de reavaliação com dados válidos.
- Validar bloqueio de envio quando campos obrigatórios não forem preenchidos.
- Validar listagem de reavaliações por paciente.
- Validar abertura/detalhamento de uma reavaliação existente.
- Validar edição de uma reavaliação existente.
- Validar comportamento quando a API retornar erro.
- Validar estados de loading durante carregamento e envio dos dados.
- Validar se o ID do paciente está sendo enviado corretamente nas operações.
- Validar responsividade básica da tela em desktop e mobile.

## Observações técnicas

- A implementação deve acompanhar o contrato final disponibilizado pelo backend na issue [#54](https://github.com/FabioCarlesso/carlessopilatesapi/issues/54).
- Caso os endpoints ainda não estejam finalizados, preparar a estrutura do frontend deixando a integração centralizada no service para facilitar ajustes posteriores.
- Utilizar interfaces TypeScript para tipar os dados de reavaliação.
- Avaliar reaproveitamento de componentes já existentes no prontuário, avaliação fisioterapêutica ou plano de tratamento.
- Manter a reavaliação como histórico, sem substituir a avaliação inicial do paciente.
- Futuramente esse módulo pode ser utilizado para relatórios comparativos de evolução do paciente.

## Labels sugeridas

- `enhancement`
- `frontend`
- `angular`
- `reavaliacao`
- `prontuario`

### PR DESCRIPTION
**Pull request by FabioCarlesso**
## Resumo

- Cria model `ReavaliacaoRequestDTO`, `ReavaliacaoUpdateDTO` e `ReavaliacaoResponseDTO` em `src/app/core/models/reavaliacao.ts`
- Cria `ReavaliacaoService` com `listarPorPaciente`, `buscar`, `criar` e `atualizar` via `/api/reavaliacoes`
- Implementa `PacienteReavaliacaoListComponent`: listagem de reavaliações com detalhes inline e link para edição
- Implementa `PacienteReavaliacaoFormComponent`: formulário reativo para criação e edição, com validação de `dataReavaliacao` (obrigatório), campos opcionais e proteção de pertencimento ao paciente
- Adiciona 3 novas rotas em `app.routes.ts` antes de `pacientes/:id`: `/reavaliacoes`, `/reavaliacoes/nova`, `/reavaliacoes/:id/editar`
- Adiciona link de Reavaliações no detalhe do paciente
- Atualiza `docs/context.md` e `README.md` com nova seção, rotas e contratos de API

## Motivo da mudança

Implementa a issue #67 — módulo de Reavaliações do paciente — para integração com os endpoints do backend definidos na issue FabioCarlesso/carlessopilatesapi#54.

## Testes executados

- `npm test`: **420/420 testes passando** (incluindo 22 novos testes: 4 no service, 8 no list component e 10 no form component)
- `npm run build`: build de produção sem erros

## Arquivos principais alterados

- `src/app/core/models/reavaliacao.ts` _(novo)_
- `src/app/core/services/reavaliacao.service.ts` _(novo)_
- `src/app/core/services/reavaliacao.service.spec.ts` _(novo)_
- `src/app/pages/pacientes/paciente-reavaliacao-list/` _(novo — 4 arquivos)_
- `src/app/pages/pacientes/paciente-reavaliacao-form/` _(novo — 4 arquivos)_
- `src/app/app.routes.ts` — 3 novas rotas
- `src/app/app.routes.spec.ts` — spec atualizado com novas rotas e precedência
- `src/app/pages/pacientes/paciente-detail/paciente-detail.component.html` — link de Reavaliações
- `docs/context.md` e `README.md` — documentação atualizada

Closes #67

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by FabioCarlesso — COMMENTED** [review]
I reviewed the PR across documentation, bugs/edge cases, security, performance, Angular patterns, API integration, UI/UX, tests, and code quality. Documentation is updated and generally aligned with the new module; tests and build pass locally. I left two inline findings for edge-case/UI correctness.

File: src/app/pages/pacientes/paciente-reavaliacao-form/paciente-reavaliacao-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+              ajustesPlanoTratamento: formFields.ajustesPlanoTratamento ?? '',
+              observacoesGerais: formFields.observacoesGerais ?? ''
+            });
+          } else {
+            this.reavaliacao = null;
+          }
+          this.loading = false;
+        },
+        error: () => {
+          this.erro = 'Erro ao carregar dados da reavaliação.';
```
Comment (FabioCarlesso) [inline src/app/pages/pacientes/paciente-reavaliacao-form/paciente-reavaliacao-form.component.ts:124]: - Problem: When the initial `forkJoin` fails, the component only sets `erro` and `loading = false`; it does not mark the route/form state as invalid. Because the template renders the form whenever `!loading && !parametroInvalido`, an edit URL like `/pacientes/10/reavaliacoes/1/editar` can show an empty form after `GET /reavaliacoes/1` fails. Saving that form then goes through the create path because `this.reavaliacao` is still `null`.
- Why it matters: A transient 404/500/network failure on the edit load can accidentally create a new reavaliação for the patient instead of blocking the broken edit flow, which is a data integrity issue and confusing UX.
- Suggested improvement: Track a `carregamentoFalhou`/`formDisponivel` state or set `parametroInvalido = true` for unrecoverable load failures in edit mode, and guard `salvar()` so edit mode requires a loaded `reavaliacao`. Add a test where `reavaliacaoService.buscar` fails and assert that `criar` is not called from the edit route.
- Severity: Medium

File: src/app/pages/pacientes/paciente-reavaliacao-list/paciente-reavaliacao-list.component.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+<div *ngIf="!loading && reavaliacoes.length === 0 && !erro" class="empty-state">
+  Nenhuma reavaliação cadastrada.
+</div>
+
+<div *ngIf="reavaliacoes.length > 0" class="reavaliacoes-list">
+  <div *ngFor="let reavaliacao of reavaliacoes" class="reavaliacao-card">
+    <div class="reavaliacao-header">
+      <div class="reavaliacao-info">
+        <span class="reavaliacao-data">{{ reavaliacao.dataReavaliacao | date:'dd/MM/yyyy' }}</span>
```
Comment (FabioCarlesso) [inline src/app/pages/pacientes/paciente-reavaliacao-list/paciente-reavaliacao-list.component.html]: - Problem: `dataReavaliacao` is a date-only API value (`YYYY-MM-DD`), but Angular `DatePipe` parses that string as a JavaScript date at UTC midnight before formatting in the browser timezone. In `America/Sao_Paulo`, `2026-05-10` formats as `09/05/2026`.
- Why it matters: The list can display the clinical reavaliação date one day earlier than the date saved in the form/API, which is a user-visible data accuracy bug.
- Suggested improvement: Format local-date strings without converting through `Date`, for example with a shared `formatLocalDate` helper/pipe that splits `YYYY-MM-DD`, or pass an explicit UTC timezone consistently if that matches the product expectation. Add a component/template test covering a `YYYY-MM-DD` value under the app timezone.
- Severity: Medium

File: src/app/pages/pacientes/paciente-reavaliacao-form/paciente-reavaliacao-form.component.ts (hunk 1 above)
Comment (FabioCarlesso) [reply src/app/pages/pacientes/paciente-reavaliacao-form/paciente-reavaliacao-form.component.ts:124]: Corrigido no commit f6599ff. Agora falhas no carregamento inicial marcam o estado como inválido e `salvar()` também bloqueia explicitamente a edição quando a reavaliação não foi carregada. Adicionei teste cobrindo falha em `reavaliacaoService.buscar` na rota de edição e garantindo que nem `criar` nem `atualizar` sejam chamados.

File: src/app/pages/pacientes/paciente-reavaliacao-list/paciente-reavaliacao-list.component.html (hunk 2 above)
Comment (FabioCarlesso) [reply src/app/pages/pacientes/paciente-reavaliacao-list/paciente-reavaliacao-list.component.html]: Corrigido no commit f6599ff. Troquei o `DatePipe` por um `LocalDatePipe` standalone para valores `YYYY-MM-DD`, sem conversão por `Date`, e adicionei testes unitários do pipe e teste de template garantindo que `2026-05-10` seja exibido como `10/05/2026`.
